testplan/base.py

```diff
@@ -430,3 +431,12 @@ def test_plan_inner_inner():
 
 
 test_plan = Testplan.main_wrapper
+
+
+class TestplanMock(Testplan):
+    """A mock Testplan class for testing purpose"""
+
+    def __init__(self, *args, **kwargs):
+        kwargs["abort_signals"] = []  # mock testplan could run in threads
+        kwargs.setdefault("runpath", tempfile.mkdtemp())
```

Comment:
seems this temp dir will not automatically removed, if the test fails and we run it again and again, them more and more temp directories will be made and a lot of disk space might be occupied. Is it better get a fixed temp directory thus the test will use the same space again. or remove the temp dir in some post steps (e.g. TestplanMock can accept an augument to decide if do clear job after test, by default it is `True`) ?

Reply:
It seems we cannot have it all. The most important thing is to make sure runpath doesn't collide to ensure test stability. And I prefer that we keep the test output for easy investigation. So the best compromise is to write to /tmp and let system clean it up. I'm wrapping pytest fixture tmp_path that creates a very nice tmp dir with testcase information.